src/reducers/sources.js

```diff
@@ -284,8 +284,12 @@ export function getSourceByURL(state: OuterState, url: string) {
   return state.sources.sources.find(source => source.get("url") === url);
 }
 
+export function getSourceInSources(sources: I.Map<string, Source>, id: string) {
```

Comment:
Well the problem is we already have a selector `getSourceById` that takes the full state and an id. This function takes just the actual sources map and an id. For various kinds of callers, we need both.